### ISSUE TITLE
can't create components with the same name in a different apps 

### ISSUE THREAD

**Issue by kadel** [redhat-developer/odo#214]
steps to reproduce:
```
▶ ./ocdev component create nodejs --git https://github.com/openshift/nodejs-ex.git
--> Found image edd4e72 (7 days old) in image stream "openshift/nodejs" under tag "6" for "nodejs"

    Node.js 6
    ---------
    Node.js 6 available as docker container is a base platform for building and running various Node.js 6 applications and frameworks. Node.js is a platform built on Chrome's JavaScript runtime for easily building fast, scalable network applications. Node.js uses an event-driven, non-blocking I/O model that makes it lightweight and efficient, perfect for data-intensive real-time applications that run across distributed devices.

    Tags: builder, nodejs, nodejs6

    * A source build using source code from https://github.com/openshift/nodejs-ex.git will be created
      * The resulting image will be pushed to image stream "nodejs:latest"
      * Use 'start-build' to trigger a new build
    * This image will be deployed in deployment config "nodejs"
    * Port 8080/tcp will be load balanced by service "nodejs"
      * Other containers can access this service through the hostname "nodejs"

--> Creating resources with label app=app,app.kubernetes.io/component-name=nodejs,app.kubernetes.io/name=app ...
    imagestream "nodejs" created
    buildconfig "nodejs" created
    deploymentconfig "nodejs" created
    service "nodejs" created
--> Success
    Build scheduled, use 'oc logs -f bc/nodejs' to track its progress.
    Application is not exposed. You can expose services to the outside world by executing one or more of the commands below:
     'oc expose svc/nodejs'
    Run 'oc status' to view your app.

▶ ./ocdev app list
ACTIVE   NAME
  *      app

▶ ./ocdev app create fff
Creating application: fff
Switched to application: fff

▶ ./ocdev app list
ACTIVE   NAME
         app
  *      fff

▶ ./ocdev component create nodejs --git https://github.com/openshift/nodejs-ex.git
unable to create git component nodejs: command: [/usr/local/bin/oc new-app nodejs~https://github.com/openshift/nodejs-ex.git --name nodejs --labels app.kubernetes.io/name=fff,app=fff,app.kubernetes.io/component-name=nodejs] failed to run:
--> Found image dbbf38f (22 seconds old) in image stream "myproject/nodejs" under tag "latest" for "nodejs"

    myproject/nodejs-1:413f2f02
    ---------------------------
    Node.js 6 available as docker container is a base platform for building and running various Node.js 6 applications and frameworks. Node.js is a platform built on Chrome's JavaScript runtime for easily building fast, scalable network applications. Node.js uses an event-driven, non-blocking I/O model that makes it lightweight and efficient, perfect for data-intensive real-time applications that run across distributed devices.

    Tags: builder, nodejs, nodejs6

    * A source build using source code from https://github.com/openshift/nodejs-ex.git will be created
      * The resulting image will be pushed to image stream "nodejs:latest"
      * Use 'start-build' to trigger a new build
    * This image will be deployed in deployment config "nodejs"
    * Port 8080/tcp will be load balanced by service "nodejs"
      * Other containers can access this service through the hostname "nodejs"

--> WARNING: output image of "myproject/nodejs:latest" should be different than input
--> Creating resources with label app=fff,app.kubernetes.io/component-name=nodejs,app.kubernetes.io/name=fff ...
    error: imagestreams.image.openshift.io "nodejs" already exists
    error: buildconfigs.build.openshift.io "nodejs" already exists
    error: deploymentconfigs.apps.openshift.io "nodejs" already exists
    error: services "nodejs" already exists
--> Failed
```

We need somehow namespace component objects, for example with prefixes or something like that.

**Comment by kadel** [comment]
Proposed solution for this:
lets use following naming scheme for OpenShift objects `{applicationName}-{componentName}`.
Currently, we are naming OpenShift object with a component name. This is causing collisions.

**Comment by slemeur** [comment]
I think this issue is going to be pretty impactful to end-users. 

Would be ideal to fix it before we ship the Milestone-MVP

**Comment by cdrage** [comment]
I'm going to assign this to myself and take a stab at it. If I don't make much progress the next couple of days I'll try getting some help :+1:

**Comment by cdrage** [comment]
Hey @kadel just want your thoughts on this,

should we be "hiding" the namespaces?

I'm currently re-implementing the PR and I added most of the code under `cmd` rather than then the `application` and `component` packages.

So now It's coming up as:

```sh
github.com/redhat-developer/odo  namespace-components ✗                                                                                                                                                                                                                                                                                                              21m ⚑  
▶ ./odo url list
Found the following URLs for component  in application nodeapp:
nodeapp-nodejs - http://nodeapp-nodejs-myproject.192.168.42.85.nip.io

github.com/redhat-developer/odo  namespace-components ✗                                                                                                                                                                                                                                                                                                              21m ⚑  
▶ ./odo list
You have deployed:
nodeapp-nodejs using the nodejs component
```

It's a *much* easier implementation, but of-course, the "application name" is being listed when doing anything to do with list.

Should we hide the applicationName instead, implement the namespacing logic lower down in the packages at the `component` and `application` levels.

**Comment by kadel** [comment]
Yes, we should hide it.
Uses should be exposed just to component names.

Internally every OpenShift object that we create has `app.kubernetes.io/component-name` to store component name and `app.kubernetes.io/name`   to store app name. 
I was thinking that we should use those labels for most of the operations that we do on OpenShift object, so we don't have to really on actual object names.

**Comment by kadel** [comment]
closed in #480
